test/subspace-test-service/src/lib.rs

```diff
@@ -901,16 +901,20 @@ where
 /// Produce the given number of blocks for both the primary node and the domain nodes
 #[macro_export]
 macro_rules! produce_blocks {
-    ($primary_node:ident, $($domain_node:ident),+, $count: literal) => {
-        async {
-            let domain_fut = {
-                let mut futs: Vec<std::pin::Pin<Box<dyn futures::Future<Output = ()>>>> = Vec::new();
-                $( futs.push( Box::pin( $domain_node.wait_for_blocks($count) ) ); )+
-                futures::future::join_all(futs)
-            };
-            $primary_node.produce_blocks($count).await?;
-            domain_fut.await;
-            Ok::<(), Box<dyn std::error::Error>>(())
+    ($primary_node:ident, $operator_node:ident, $count: literal $(, $domain_node:ident)*) => {
+        {
+            let _ = $operator_node.send_remark_extrinsic().await;
+            async {
+                let domain_fut = {
+                    let mut futs: Vec<std::pin::Pin<Box<dyn futures::Future<Output = ()>>>> = Vec::new();
+                    futs.push(Box::pin($operator_node.wait_for_blocks(1)));
+                    $( futs.push( Box::pin( $domain_node.wait_for_blocks($count) ) ); )*
+                    futures::future::join_all(futs)
+                };
+                $primary_node.produce_blocks($count).await?;
+                domain_fut.await;
+                Ok::<(), Box<dyn std::error::Error>>(())
+            }
```

Comment:
This macro may produce multiple blocks thus we should send the same number of extrinsic instead of just one, perhaps something like:
```
async {
    for _ in 0..$count {
        let _ = $operator_node.send_remark_extrinsic().await;
        ...
        $( futs.push( Box::pin( $domain_node.wait_for_blocks(1) ) ); )*
        ...
        $primary_node.produce_blocks(1).await?;
    }
}
```

Reply:
Not required since the confirmation should kick. We can clean it up later. This is good enough IMHO